students/DiannaTingg/Session04/trigrams.py

```diff
@@ -0,0 +1,102 @@
+# Lesson 04 Exercise: Trigrams
+import sys
+import random
+
+
+# Make a dictionary with the first two words as the key and the third word as the value
+def make_trigrams(words):
+    tris = {}
+
+    for i in range(len(words)-2):
+        pair = words[i:i+2]
+        follower = words[i+2]
+
+        # Convert to a tuple
+        pair = (pair[0], pair[1])
+
+        # If in dictionary, append value, otherwise add it
+        if pair in tris:
+            tris[pair].append(follower)
+        else:
+            tris[pair] = [follower]
+    return tris
+
+
+# Get random word from dictionary
+def get_random_word(d):
+    word = random.choice(list(d.keys()))
+    return word
+
+
+# Use dictionary to build new text
+def use_trigrams(d):
+    start = get_random_word(d)
+
+    alternate_word = ""
+
+    text_words = [start[0], start[1]]
+
+    while len(text_words) < 250:
```

Comment:
Getting hung up on this function when running this.  Could be the text file I'm using.  This while loop appears to be the culprit.  Let's discuss.